nym-api/src/support/http/mod.rs

```diff
@@ -173,27 +245,24 @@ pub(crate) async fn setup_routes(network_monitor: bool) -> anyhow::Result<Router
         //         .on_request(DefaultOnRequest::new())
         //         .on_response(DefaultOnResponse::new().latency_unit(tower_http::LatencyUnit::Micros)),
         // )
-        .route(
-            "/",
-            axum::routing::get(|| async { axum::response::Redirect::permanent("/swagger") }),
+        // .route("/", axum::routing::get(|| async {axum::response::Redirect::permanent("/swagger")}))
+        // .route("/swagger", axum::routing::get(hello))
+        .merge(SwaggerUi::new("/swagger").url("/api-docs/openapi.json", ApiDoc::openapi()))
+        .nest(
+            "/v1",
+            Router::new()
+                .nest("/circulating-supply", circulating_supply_routes())
+                .merge(nym_contract_cache_routes())
+                .nest("/status", node_status_routes(network_monitor))
+                .nest("/network", nym_network_routes())
+                .nest("/api-status", status::handlers::api_status_routes())
+                .merge(nym_node_routes())
+                .nest("/unstable/nym-nodes", nym_node_routes_unstable()), // CORS layer needs to be "outside" of routes
         )
-        .route("/swagger", axum::routing::get(hello))
-        .merge(circulating_supply_routes())
-        .merge(nym_contract_cache_routes())
-        .merge(node_status_routes(network_monitor))
-        .merge(nym_network_routes())
-        .merge(status::handlers::api_status_routes())
-        .merge(nym_node_routes())
-        .merge(nym_node_routes_unstable())
-        // CORS layer needs to be "outside" of routes
         .layer(setup_cors());
 
     // needs to be after routes we want to log
     let router = router.layer(axum::middleware::from_fn(logger));
```

Comment:
nitpicking: just layer the logger middleware when you're creating the original `Router`

Reply:
this is intentional: logging needs to be outermost layer. In fact, I'm working on a design that will ensure logging and CORS layers are outermost and nobody can accidentally add a layer in the wrong place

Reply:
aha! makes sense, good thinking!